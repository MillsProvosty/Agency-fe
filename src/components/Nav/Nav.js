import React from "react";
import "./Nav.scss";
import { FaHandsHelping } from "react-icons/fa";
import { NavLink, Link } from "react-router-dom";

const Nav = () => {
  return (
    <section className="nav">
      <NavLink to="/about" style={{ textDecoration: 'none' }}>
        <section className="logo">
          <FaHandsHelping size={64} />
          <p className="title">Agency</p>
        </section>
      </NavLink>
      <section className="links">
        <NavLink to="/about" className="dash-link">
          About
        </NavLink>
        <NavLink to="/profile" className="dash-link">
          Profile
        </NavLink>
        <NavLink to="/schedule" className="dash-link">
          Schedule
        </NavLink>
        <NavLink to="/tasks" className="dash-link">
          Tasks
        </NavLink>
        <NavLink to="/history" className="dash-link">
          History
        </NavLink>
        <NavLink to="/team" className="dash-link">
          Team
        </NavLink>
      </section>
    </section>
  );
};

export default Nav;
