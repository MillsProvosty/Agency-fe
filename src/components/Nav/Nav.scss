.nav {
  display: flex;
  justify-content: space-between;
  background-color: darkblue;
  color: white;
  height: 100px;

  & .logo {
    color: white;

    & .title {
      text-decoration: none;
    }

    & .title:hover {
      border-bottom: 1px solid white;
      height: 1.25em;
    }
  }

  & .links {
    width: 60%;
    display: flex;
    justify-content: space-evenly;

    & .dash-link {
      text-decoration: none;
      font-size: 1.5em;
      color: white;
      margin-top: 1.5em;
    }

    & .dash-link:hover {
      border-bottom: 1px solid white;
      height: 1.5em;
    }
  }

  & .logo {
    display: flex;
    margin: 20px;
    font-size: 1.5em;

    & p {
      margin-left: 10px;
    }
  }
}


@media screen and (max-width: 375px) {
  .nav {
    & .nav-link {
      display: none;
    }

    & .links {
      & .dash-link {
        font-size: 1em;
        margin-top: 2.5em;
        padding: 5px;
      }
    }
  }
}