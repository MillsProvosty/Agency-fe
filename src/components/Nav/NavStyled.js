import styled from "styled-components";

export const NavBar = styled.section`
  display: flex;
  justify-content: space-between;
  background-color: #37474e;
  color: aliceblue;
  height: 100px;
`;

export const LogoSection = styled.section`
  color: aliceblue;
  display: flex;
  margin: 10px;
  font-size: 1.5em;
  align-items: center;
`;

export const PTag = styled.p`
  font-size: 2em;
  margin: 0;
`;

export const PLinks = styled.p`
  font-size: 1.5em;
  color: aliceblue;
  margin-top: 1.5em;
`;

export const Logo = styled.button`
  background-color: #37474e;
  border-radius: 50%;
  border: 2px solid white;
`;

export const LinksSection = styled.section`
  width: 60%;
  display: flex;
  justify-content: space-evenly;
`;
