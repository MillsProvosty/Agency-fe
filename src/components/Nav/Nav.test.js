import React from 'react';
import ReactDOM from 'react-dom';
import Nav from './Nav';

it('renders without crashing', () => {
  expect(true).toEqual(true)
});
