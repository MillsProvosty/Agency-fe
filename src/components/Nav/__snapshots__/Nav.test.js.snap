// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match snapshot 1`] = `
<styled.section
  className="nav"
>
  <NavLink
    className="nav-link"
    style={
      Object {
        "textDecoration": "none",
      }
    }
    to="/about"
  >
    <styled.section
      className="logo"
    >
      <styled.button>
        <GiAirBalloon
          className="hands"
          size={40}
          style={
            Object {
              "color": "white",
            }
          }
        />
      </styled.button>
      <styled.p
        className="title"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        Agency
      </styled.p>
    </styled.section>
  </NavLink>
  <styled.section
    className="links"
  >
    <NavLink
      className="dash-link"
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/about"
    >
      <styled.p>
        About
      </styled.p>
    </NavLink>
    <NavLink
      className="dash-link"
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/profile"
    >
      <styled.p>
        Profile
      </styled.p>
    </NavLink>
    <NavLink
      className="dash-link"
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/schedule"
    >
      <styled.p>
        Schedule
      </styled.p>
    </NavLink>
    <NavLink
      className="dash-link"
      style={
        Object {
          "textDecoration": "none",
        }
      }
      to="/team"
    >
      <styled.p>
        Team
      </styled.p>
    </NavLink>
  </styled.section>
</styled.section>
`;
